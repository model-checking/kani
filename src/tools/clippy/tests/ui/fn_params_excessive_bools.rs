#![warn(clippy::fn_params_excessive_bools)]
#![allow(clippy::too_many_arguments)]

extern "C" {
    fn f(_: bool, _: bool, _: bool, _: bool);
}

macro_rules! foo {
    () => {
        fn fff(_: bool, _: bool, _: bool, _: bool) {}
    };
}

foo!();

#[no_mangle]
extern "C" fn k(_: bool, _: bool, _: bool, _: bool) {}
fn g(_: bool, _: bool, _: bool, _: bool) {}
fn h(_: bool, _: bool, _: bool) {}
fn e(_: S, _: S, _: Box<S>, _: Vec<u32>) {}
fn t(_: S, _: S, _: Box<S>, _: Vec<u32>, _: bool, _: bool, _: bool, _: bool) {}

struct S {}
trait Trait {
    fn f(_: bool, _: bool, _: bool, _: bool);
    fn g(_: bool, _: bool, _: bool, _: Vec<u32>);
}

impl S {
    fn f(&self, _: bool, _: bool, _: bool, _: bool) {}
    fn g(&self, _: bool, _: bool, _: bool) {}
    #[no_mangle]
    extern "C" fn h(_: bool, _: bool, _: bool, _: bool) {}
}

impl Trait for S {
    fn f(_: bool, _: bool, _: bool, _: bool) {}
    fn g(_: bool, _: bool, _: bool, _: Vec<u32>) {}
}

fn main() {
    fn n(_: bool, _: u32, _: bool, _: Box<u32>, _: bool, _: bool) {
        fn nn(_: bool, _: bool, _: bool, _: bool) {}
    }
}
