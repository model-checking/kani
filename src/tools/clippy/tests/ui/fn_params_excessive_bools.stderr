error: more than 3 bools in function parameters
  --> $DIR/fn_params_excessive_bools.rs:18:1
   |
LL | fn g(_: bool, _: bool, _: bool, _: bool) {}
   | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
   |
   = note: `-D clippy::fn-params-excessive-bools` implied by `-D warnings`
   = help: consider refactoring bools into two-variant enums

error: more than 3 bools in function parameters
  --> $DIR/fn_params_excessive_bools.rs:21:1
   |
LL | fn t(_: S, _: S, _: Box<S>, _: Vec<u32>, _: bool, _: bool, _: bool, _: bool) {}
   | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
   |
   = help: consider refactoring bools into two-variant enums

error: more than 3 bools in function parameters
  --> $DIR/fn_params_excessive_bools.rs:25:5
   |
LL |     fn f(_: bool, _: bool, _: bool, _: bool);
   |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
   |
   = help: consider refactoring bools into two-variant enums

error: more than 3 bools in function parameters
  --> $DIR/fn_params_excessive_bools.rs:30:5
   |
LL |     fn f(&self, _: bool, _: bool, _: bool, _: bool) {}
   |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
   |
   = help: consider refactoring bools into two-variant enums

error: more than 3 bools in function parameters
  --> $DIR/fn_params_excessive_bools.rs:42:5
   |
LL | /     fn n(_: bool, _: u32, _: bool, _: Box<u32>, _: bool, _: bool) {
LL | |         fn nn(_: bool, _: bool, _: bool, _: bool) {}
LL | |     }
   | |_____^
   |
   = help: consider refactoring bools into two-variant enums

error: more than 3 bools in function parameters
  --> $DIR/fn_params_excessive_bools.rs:43:9
   |
LL |         fn nn(_: bool, _: bool, _: bool, _: bool) {}
   |         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
   |
   = help: consider refactoring bools into two-variant enums

error: aborting due to 6 previous errors

